### PR TITLE
Refactor Pfam data persistence

### PR DESCRIPTION
**Pull request by matthiasblum**
- Implement a generic Stockholm format parser to extract relevant information from:
  - `Pfam-A.seed.gz`
  - `Pfam-A.full.gz`
  - `Pfam-C.gz`
- Remove code related to UniProt/RP15/RP35/RP55/RP75, which are no longer supported in the InterPro website.
- Retrieve alignments directly from `Pfam-A.seed.gz` and `Pfam-A.full.gz` instead of individual files.
- Simplify persistence by dumping all required data into two tables: `PFAM_A` (families) and `PFAM_C` (clans). Alignments are stored gzip-compressed.

`PFAM_A` and `PFAM_C` tables created on `IPTST` (using Pfam 37.0 data).

### CODE REVIEW

**Review by HobnobMancer — CHANGES_REQUESTED** [review]
Generally looks good :smiley: 
There's a couple of issues with handling missing data, which I've flagged

File: pyinterprod/interpro/contrib/pfam.py [hunk 1]
```diff
@@ -631,358 +90,378 @@ def get_protenn_entries(cur, file: str) -> list[Method]:
...
-    Persist additional Pfam data from Pfam release files that is not captured in the signatures
-        or clan tables. Primarily persists data required by interpro7.
+def get_signatures(pfam_seed_file: str) -> list[Method]:
+    methods = []
+    for entry in parse_sto(pfam_seed_file):
+        comment = entry.features.get("CC")
+        rn2pmid = {}
+        if comment:
+            for i, obj in enumerate(entry.features.get("RN", [])):
+                rn2pmid[i+1] = int(obj["RM"])
```
[the remark this answers is not in the document]
Comment (matthiasblum) [reply pyinterprod/interpro/contrib/pfam.py:103]: Using ` int(" ".join(obj["RM"]))` so a ValueError is raised when there are two PMIDs or more.

File: pyinterprod/interpro/clan.py [hunk 2]
```diff
@@ -620,20 +617,14 @@ def update_hmm_clans(
...
-
-        def getsubdir(x): return x[:7]
-    elif database.name.lower() == "pfam":
-        clans = contrib.pfam.get_clans(
-            pfam_clan_path,
-            pfam_fasta_path,
-            pfam_full_path,
-        )
+    if database.name.lower() == "pfam":
+        clans = contrib.pfam.get_clans(kwargs["clan"],
```
Comment (laisecf) [inline pyinterprod/interpro/clan.py:621]:
```suggestion
        clans = contrib.pfam.get_clans(kwargs["clans"],
```

**Review by HobnobMancer — APPROVED** [review]
LGTM

**Review by laisecf — APPROVED** [review]
Great refactoring!! 🎉  Everything running perfectly in the latest tests

**Pull request merged**